apps/emqx_management/src/emqx_mgmt_api_listeners.erl

```diff
@@ -251,6 +262,18 @@ fields(node_status) ->
             })},
         {status, ?HOCON(?R_REF(status))}
     ];
+fields({Type, with_name}) ->
+    Schema = fields(Type),
+    lists:keyreplace(
+        id,
+        1,
+        Schema,
+        {name,
+            ?HOCON(binary(), #{
+                desc => "Listener name",
+                required => true
+            })}
+    );
```

Comment:
`fields/1` is the callback for all defined structs,
since this clause is only used locally in this module: 
please move this to a internal function instead of reusing the `fields/1` callback.

Reply:
This is not a direct type, but a macro-like type, generated by the code, and  the result is different with the old `fields(Type)` ,  just like a  variant type of the `fiels(Type)`, so I wrote it as so.